### PR TITLE
ENGDESK-7906: Add new originate variable originate_request_timeout

### PR DESCRIPTION
**Pull request by dev-ryanc**

### CODE REVIEW

File: src/switch_ivr_originate.c [hunk 1]
```diff
@@ -3293,6 +3309,14 @@ SWITCH_DECLARE(switch_status_t) switch_ivr_originate(switch_core_session_t *sess
 					switch_ivr_parse_all_events(oglobals.session);
 				}
 
+				if (global_elapsed > (time_t) request_timelimit_sec) {
+					oglobals.idx = IDX_TIMEOUT;
+					if (force_reason == SWITCH_CAUSE_NONE) {
+						force_reason = SWITCH_CAUSE_PROGRESS_TIMEOUT;
```
Comment (pchero) [inline src/switch_ivr_originate.c:3315]: Could we introduce new force_reason something like `SWITH_CAUSE_GPROGRESS_TIMEOUT`?
May we will need to distinguish which time out caused call hangup in the future..

Comment (pchero) [reply src/switch_ivr_originate.c:3315]: Hm.. I think adding a new variable is out of this scope. I think this should be OK for now.

**Review by pchero — APPROVED** [review]
Good! Looks OK to me. :)

**Review by StefanYohansson — APPROVED** [review]

**Review by pchero — APPROVED** [review]
Looks good to me.

**Pull request merged**